mergify_engine/context.py

```diff
@@ -422,24 +415,68 @@ async def get_mergify_config_file(self) -> typing.Optional[MergifyConfigFile]:
         if mergify_config_file is not cache.Unset:
             return mergify_config_file
 
-        config_location_cache = self.get_config_location_cache_key(
-            self.installation.owner_login, self.repo["name"]
+        cached_config_file = await self.get_cached_config_file(
+            self.repo["id"],
         )
-        cached_filename = await self.installation.redis.get(config_location_cache)
+
+        if cached_config_file is not None:
+            self._caches.mergify_config_file.set(cached_config_file)
+            return cached_config_file
+
+        config_file_cache_key = self.get_config_file_cache_key(self.repo["id"])
+        config_location_cache_key = self.get_config_location_cache_key(self.repo["id"])
+        cached_filename = await self.installation.redis.get(config_location_cache_key)
+        pipeline = await self.installation.redis.pipeline()
+
         async for config_file in self.iter_mergify_config_files(
             preferred_filename=cached_filename
         ):
+
             if cached_filename != config_file["path"]:
-                await self.installation.redis.set(
-                    config_location_cache, config_file["path"], ex=60 * 60 * 24 * 31
+                await pipeline.set(
+                    config_location_cache_key, config_file["path"], ex=60 * 60 * 24 * 31
                 )
+
+            await pipeline.set(
+                config_file_cache_key,
+                json.dumps(
+                    {
+                        "type": config_file["type"],
+                        "content": config_file["content"],
+                        "path": config_file["path"],
+                        "sha": config_file["sha"],
+                    }
```

Comment:
It'd be better to have a `serialize` method on the `MergifyConfigFile` that does this.